lib/floki/selector/pseudo_class.ex

```diff
@@ -45,6 +45,31 @@ defmodule Floki.Selector.PseudoClass do
     false
   end
 
+  def match_nth_of_type?(_, %HTMLNode{parent_node_id: nil}, _), do: false
+  def match_nth_of_type?(tree, html_node, %__MODULE__{value: -1}) do
+    children_nodes_ids = get_children_nodes_indexed_by_type(tree, html_node.parent_node_id, html_node.type)
```

Comment:
Since you are accessing two attributes of the same record, I would recommend to only pass the `html_node` to the `get_children_nodes_indexed_by_type` function (changing the arity from 3 to 2).

Reply:
I initially passed the `parent_node_id` and `type` separately to be consistent with `get_children_nodes_indexed/2` but you're right, it makes sense to pass `html_node` as a whole. I've pushed a commit which should address this. Thanks.